src/amo/components/SponsoredAddonsShelf/index.js

```diff
@@ -29,17 +36,40 @@ type Props = {|
 
 export type InternalProps = {|
   ...Props,
+  _config: typeof config,
   _tracking: typeof tracking,
+  dispatch: DispatchFunc,
+  errorHandler: ErrorHandlerType,
   i18n: I18nType,
+  isLoading: boolean,
   resultsLoaded: boolean,
+  shelfData: SponsoredShelfType | null | void,
   shelves: { [shelfName: string]: Array<AddonType> | null },
 |};
 
 export class SponsoredAddonsShelfBase extends React.Component<InternalProps> {
   static defaultProps = {
+    _config: config,
     _tracking: tracking,
+    shelfData: undefined,
   };
 
+  componentDidMount() {
+    const { _config, isLoading, shelfData } = this.props;
+
+    if (
+      _config.get('enableFeatureUseAdzerkForSponsoredShelf') &&
+      !isLoading &&
+      !shelfData
+    ) {
+      this.props.dispatch(
+        fetchSponsored({
+          errorHandlerId: this.props.errorHandler.id,
+        }),
+      );
+    }
+  }
```

Comment:
We tend to use the `constructor` for this kind of logic. Part of the reason is (I think) that `componentDidMount` isn't called on the server.